TFEngine.py

```diff
@@ -1889,15 +1892,13 @@ def extra_fetches_callback(seq_idx, seq_tag, output, targets=None, beam_scores=N
       runner.num_steps, self.format_score(runner.score), self.format_score(runner.error)), file=log.v1)
     if output_file:
       assert out_cache
-      assert 0 in out_cache
-      assert len(out_cache) - 1 in out_cache
```

Comment:
Why? Esp for the txt output format, if some seqs are missing, there is no way to see that, or see which seqs are missing. Also, usually all further tools would expect that the seqs are complete.

Reply:
Ok, the question is whether there is a use case for filtering sentences by length when doing search. If no, we should assert max_seq_length == 0. If yes, then the output file inevitably will not contain all sequences.

As I said, the behaviour right now is that filtered source sequences are not put into any batch, therefore not translated, therefore not appearing in out_cache.  When we then loop from 0 to len(out_cache), there will be a key error. (This is not caught by the asserts.)

Reply:
Yes. I think an `assert max_seq_length == 0` might be better (for now). Or leave the warning as you did, and then move the `assert 0 in out_cache`  and  the other for only the txt output format. But maybe just do `assert max_seq_length == 0`. I cannot really think of a case where it would be a good idea to ignore some seqs.

Reply:
Ok, I'll keep the assertions.
How about setting max_seq_length = 0 and printing a warning that max_seq_length from config was ignored? Because I know that some people use max_seq_length = 1000 or so for search, which would fail if only 0 is allowed. Or should we force them to use the right value? ;)

Reply:
Hm, not sure. Maybe just force them. Because maybe at some point we want to support that (`max_seq_length`), and then do some more clever thing (whatever that could be, maybe configurable, like translating a subseq only, or just split it, or so).